src/icon-loader.c

```diff
@@ -135,6 +137,33 @@ process_abs_name(struct icon_ctx *ctx, const char *icon_name)
 	return -1;
 }
 
+/*
+ * Looks up an application desktop entry using fuzzy matching
+ * (e.g. "thunderbird" matches "org.mozilla.Thunderbird.desktop"
+ * and "XTerm" matches "xterm.desktop"). This is not per any spec
+ * but is needed to find icons for existing applications.
+ */
+static struct sfdo_desktop_entry *
+get_db_entry_by_id_fuzzy(struct sfdo_desktop_db *db, const char *app_id)
+{
+	size_t n_entries;
+	struct sfdo_desktop_entry **entries = sfdo_desktop_db_get_entries(db, &n_entries);
+
+	for (size_t i = 0; i < n_entries; i++) {
+		struct sfdo_desktop_entry *entry = entries[i];
+		const char *desktop_id = sfdo_desktop_entry_get_id(entry, NULL);
+		/* Get portion of desktop ID after last '.' */
+		const char *dot = strrchr(desktop_id, '.');
+		const char *desktop_id_base = dot ? (dot + 1) : desktop_id;
+
+		if (g_ascii_strcasecmp(app_id, desktop_id_base) == 0) {
```

Comment:
Prefer `!strcasecmp()`

Reply:
My concerns with strcasecmp is the locale-specific behavior - it is "implementation defined" in all but the C/POSIX locale. But maybe it's not an issue in practice.

Reply:
Okay. Let's leave as is.

Reply:
Ah... sorry you've changed it :smile:

Reply:
Oh, I already switched it to strcasecmp. Either way is fine with me.